nav2_lifecycle_manager/src/lifecycle_manager.cpp

```diff
@@ -126,29 +134,38 @@ LifecycleManager::destroyLifecycleServiceClients()
   }
 }
 
-void
+bool
+LifecycleManager::changeStateForNode(const std::string & node_name, std::uint8_t transition)
+{
+  if (!node_map_[node_name]->change_state(transition) ||
+    !(node_map_[node_name]->get_state() == transition_state_map_[transition]))
+  {
+    RCLCPP_ERROR(get_logger(), "Failed to change state for node: %s", node_name.c_str());
+    return false;
```

Comment:
This is one of those places where throwing an exception would be better. Notice how you just need to keep checking the result and passing it back up the stack. All the intermediate functions would be simpler using exceptions.

Reply:
This would work better still if `change_state` threw an exception.

Reply:
So we would catch the exception in the intermediate functions (e.g. bringupNode)? RIght now I don't think we're doing anything except reporting a failed transition